### ISSUE TITLE
indentation of conditional function definition

### ISSUE THREAD

**Issue by mmaechler** [emacs-ess/ESS#337]
As usually, the problem is not easily visible in the proportional fonts inside github.
What I'd want is that  the outermost  `else` aligns with its corresponding `if` .. but it is indented **less**.
This is even more the case if wrap the two lines  `tfun <-  \\  if(is.function....)` into a single line.

```r
## from a robustbase vignette:
{
    ## calculate robustness weights
    lwgts <- Mwgt(lresid, lctrl$tuning.psi, lctrl$psi)
    ## function to calculate robustified leverages
    tfun <-
        if (is.function(attr(estlist$design, 'gen')))
            function(i) {
                if (all(is.na(wi <- lwgts[i,]))) wi
                else .lmrob.hat(lXs[,,i,lcdn[2]],wi)
            }
    else
        function(i) {
            if (all(is.na(wi <- lwgts[i,]))) wi else .lmrob.hat(lX, wi)
        }
}
```

**Comment by wmay** [comment]
Does that even run? I can't use `else` statements on new lines without wrapping the `if` block in curly brackets. It indents fine for me with the brackets added.

**Comment by lionel-** [comment]
I'll hopefully fix these issues once and for all when the new parser stuff is finished. I'll try to find some time to work on it before I hand in my thesis, it's long overdue.

**Comment by vspinu** [comment]
Yes. No hurry. It's a corner case with easy workaround anyways. I just decided to try "assignee" feature on github ;)
